File: app/src/main/java/com/kickstarter/libs/preferences/BooleanPreferenceType.java
package com.kickstarter.libs.preferences;

public interface BooleanPreferenceType {
  /**
   * Get the current value of the preference.
   */
  boolean get();

  /**
   * Returns whether a value has been explicitly set for the preference.
   */
  boolean isSet();

  /**
   * Set the preference to a value.
   */
  void set(boolean value);

  /**
   * Delete the currently stored preference.
   */
  void delete();
}

File: app/src/main/java/com/kickstarter/libs/preferences/BooleanPreferenceType.kt
package com.kickstarter.libs.preferences

interface BooleanPreferenceType {
    /**
     * Get the current value of the preference.
     */
    fun get(): Boolean

    /**
     * Returns whether a value has been explicitly set for the preference.
     */
    val isSet: Boolean

    /**
     * Set the preference to a value.
     */
    fun set(value: Boolean)

    /**
     * Delete the currently stored preference.
     */
    fun delete()
}
